added actions in self.actions

simusud opened the pull request:
added some setters in dex
added vote actions ,methods  and interfaces on governance
added setter for retirement bonus in loans

simusud commented:
I have added some methods from "Modefy definfVote" PR which is yet to merge . So assuming those methods will be there in governance.py after it is merged i added these methods (setVoteDuration, setQuorum, setVoteDefinitionFee, setBalnVoteDefinitionCriterion).

dbrehmer commented on core_contracts/governance/data_objects.py:
```diff
@@ -35,7 +35,19 @@ def __init__(self, db: IconScoreDatabase, gov: IconScoreBase) -> None:
             'update_mining_ratio': self._gov.setMiningRatio,
             'update_locking_ratio': self._gov.setLockingRatio,
-            'update_origination_fee': self._gov.setOriginationFee
+            'update_origination_fee': self._gov.setOriginationFee,
```
Let's make all the keys here the same as the methods. People are getting confused when they are different.

simusud commented:
i have changed action keys to same as method name in data_option,
added set minimum vote duration in governance on_install ,
 added checks for address type in addNewDataSource method
updated governance unit tests

dbrehmer commented on core_contracts/governance/data_objects.py:
```diff
@@ -30,12 +30,24 @@ def __init__(self, db: IconScoreDatabase, gov: IconScoreBase) -> None:
-            'update_mining_ratio': self._gov.setMiningRatio,
-            'update_locking_ratio': self._gov.setLockingRatio,
-            'update_origination_fee': self._gov.setOriginationFee
+            'updateBalTokenDistPercentage': self._gov.updateBalTokenDistPercentage,
```
If this will be changed (and it probably should be) we will need to change the vote actions definition for BIP5 as well.

simusud replied:
Yeah , we need to change that.

dbrehmer commented on core_contracts/governance/governance.py:
```diff
@@ -559,6 +592,8 @@ def addNewDataSource(self, _data_source_name: str, _contract_address: Address) -
         Add a new data source to receive BALN tokens. Starts with a default of
         zero percentage of the distribution.
         """
+        if type(_contract_address) != Address:
```
This may get rejected in audit. They don't like parameter types to be uncertain. Please ask about this before we submit so we can correct it if needed without losing time.

simusud replied:
Shall i remove this in this PR ?

dbrehmer reviewed (commented):
Looks good. Just two comments to check before merging. With updateDistPercent action name changing we will need to update the vote actions in the current vote. It can be a separate PR from this one though.
Also, I think it is unlikely that intentionally sending the wrong type to a method will be approved in audit. If you like you can message Jaechang to ask if he would approve it, but let's at least avoid the delay that would come with a rejection by asking first.

dbrehmer commented:
@simusud - Can we also please add the DAOfund disburse method to the vote actions.

simusud commented:
> @simusud - Can we also please add the DAOfund disburse method to the vote actions.

yeah sure!

dbrehmer commented on core_contracts/governance/governance.py:
```diff
@@ -907,3 +942,22 @@ def scoreUpdate_13(self) -> None:
+
+    def scoreUpdate_14(self):
+        """
+        Changing action keys to method names
```
I think this is left over from the previous time we fixed a proposal.

dbrehmer replied:
Okay, looks good.

dbrehmer reviewed (approved):
Looks good.

dbrehmer reviewed (approved):
Seems this went stale when another merge to development happened. Still looks good.
